extension-compose/src/main/java/com/mapbox/maps/extension/compose/MapboxMap.kt

```diff
@@ -27,12 +35,42 @@ public fun MapboxMap(
    */
   modifier: Modifier = Modifier,
   /**
-   * Defines the initialisation configurations for a [MapboxMap].
+   * Defines the initialisation configurations factory for a [MapboxMap].
    *
    * It can only be set once and not mutable after the initialisation. Mutating the [MapInitOptions]
    * during recomposition will result in a [IllegalStateException].
    */
-  mapInitOptions: MapInitOptions = MapInitOptions(LocalContext.current.applicationContext),
+  mapInitOptionsFactory: (Context) -> MapInitOptions = { context -> MapInitOptions(context) },
```

Comment:
@alexander-kulikovskii I've changed the `MapInitOptions` argument to a factory, so we can build it later and avoid errors in preview mode, as it doesn't need to be invoked.

Reply:
cool